### PR TITLE
[PLATFORM-1422]-Adding support for cmd_measure in coreos-grub

### PR DESCRIPTION
**Pull request by karandave-bc**
Adding a command in coreos-grub to measure a file into a PCR.

### CODE REVIEW

File: grub-core/loader/i386/linux.c [hunk 1]
```diff
@@ -1152,19 +1152,80 @@ grub_cmd_initrd (grub_command_t cmd __attribute__ ((unused)),
...
+
+  /* if index is invalid */
+  if( grub_errno != GRUB_ERR_NONE )
+    {
+      grub_error (GRUB_ERR_BAD_ARGUMENT, N_("filename and PCR expected"));
+      goto fail;
+    }
+
+  file = grub_file_open (argv[0]);
+  if (! file)
```
Comment (codymgriffin) [inline grub-core/loader/i386/linux.c:1180]: always use braces

File: grub-core/loader/i386/linux.c [hunk 2]
```diff
@@ -1152,19 +1152,80 @@ grub_cmd_initrd (grub_command_t cmd __attribute__ ((unused)),
...
+
+  if (!read_data)
+    {
+      grub_error (GRUB_ERR_OUT_OF_MEMORY, N_("cannot allocate data buffer"));
+      goto fail;
+    }
+
+  if (grub_file_read (file, read_data, len) != len)
+    {
+      if (!grub_errno)
```
Comment (codymgriffin) [inline grub-core/loader/i386/linux.c:1196]: definitely use braces, and make sure to keep the same indentation style (even though GNU style is the worst style)

File: grub-core/loader/i386/linux.c [hunk 3]
```diff
@@ -1152,19 +1152,80 @@ grub_cmd_initrd (grub_command_t cmd __attribute__ ((unused)),
   return grub_errno;
 }
 
+static grub_err_t
```
Comment (codymgriffin) [inline grub-core/loader/i386/linux.c:1155]: Do you have a link to where you copied this for a reference?

Comment (karandave-bc) [reply grub-core/loader/i386/linux.c:1155]: I didn't copy it from anywhere outside.
I just tried to follow the pattern of coreos/grub's other commands.

**Review by codymgriffin — APPROVED** [review]
Awesome - make sure to tag the branch after you merge this so we can trace it through (2.02.1-braincorp for example)

**Pull request merged**